src/primitives/torus.jl

```diff
@@ -0,0 +1,76 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    Torus(center, normal, major, minor)
+
+A torus centered at `center` with axis of revolution directed by 
+`normal` and with radii `major` and `minor`. 
+
+"""
+struct Torus{T} <: Primitive{3,T}
+  center::Point{3,T}
+  normal::Vec{3,T}
+  major::T
+  minor::T
+end
+
+function Torus(center::Tuple, normal::Tuple, major::T1, minor::T2) where {T1,T2}
+  T = promote_type(eltype(center), eltype(normal), T1, T2, Float32)  
+  Torus(Point(T.(center)), Vec(T.(normal)), T(major), T(minor))
+end
+
+"""
+  Torus(p1, p2, p3)
+
+The torus whose equator passes through points `p1`, `p2` and `p3` and with
+minor radius `minor`.
+"""
+function Torus(p1::Point{3}, p2::Point{3}, p3::Point{3}, minor)
+  # circumcenter of p1, p2, p3
+  n⃗ = normal(Plane(p1, p2, p3))
+  v12 = p2 - p1
+  v13 = p3 - p1
+  p12 = coordinates(p1 + v12/2)
+  p13 = coordinates(p1 + v13/2)
+  offset = coordinates(p1) ⋅ n⃗
+  A = transpose(hcat(n⃗, v12, v13))
+  b = [offset, p12 ⋅ v12, p13 ⋅ v13]
+  c = Point(inv(A) * b)
```

Comment:
from the code comment shouldn't this block of code be just `center(Sphere(p1, p2, p3))`?

Reply:
There's no unique sphere passing through three points in 3D :-)

Reply:
Should we implement `circumcenter` for a `Triangle`? For the tetrahedron we already use it, in the sphere passing through four points.

Reply:
That is a great idea. Let's add a new constructor for Triangle and then come back here to finish this PR 👍🏽 

In the meantime, this PR is missing tests for the regular sampling method too.